inbm-v5.md

```diff
@@ -0,0 +1,320 @@
+
+1. _cloud-connect_: when INBM is provisioned to connect to a `DMS` and receives update related `ota`commands from cloud.
+1. _local-host_: when INBM is provisioned to be only invoked by a `privilaged` user-space application running on the same host OS.
+
+Described below are the different data flow paths based on the provisioning modes for commands and information:
+
+### Cloud-connect data flow
+
+```mermaid
+sequenceDiagram
+  box Device Management Server
+    actor admin
+    participant DMS
+  end
+  
+  box INBM
+    participant cc as Cloud Client
+    participant inbmd as inbm-daemon
+    participant dispQ as dispatcher-queue
+    participant ota as ota-updater
+  end
+
+  box Update tool
+    participant isv as ISV tool
+  end 
+  
+  admin -->> DMS : Trigger OTA cmd
+  DMS ->> cc : mqtt/tls pub (e.g. /methods/POST/) <br/> OTA cmd
+  cc -->> inbmd : OTA cmd
+  inbmd -->> dispQ : OTA cmd
+  dispQ -->> dispQ : parse OTA cmd <br/> updater-type
+  dispQ --> ota : OTA cmd
+  ota ->> isv : update_tool_cmd <args>
+  isv ->> ota : status <OK/ERROR>
+  ota --> inbmd: status <OK/ERROR, msg>
+  inbmd -->> cc : status <OK/ERROR, msg>
+  cc ->> DMS : mqtt/tls pub (e.g. /status/)
+
+```
+
+### local-host data flow
+
+```mermaid
+sequenceDiagram
+  box sudo
+    participant sudo as Previlaged App
+  end
+  
+  box INBM
+    participant inbc as INBC
+    participant inbmd as inbm-daemon
+    participant dispQ as dispatcher-queue
+    participant ota as ota-updater
+  end
+
+  box Update tool
+    participant isv as ISV tool
+  end 
+  
+  sudo ->> inbc : Trigger OTA cmd
+  inbc -->> inbmd : unix sock: OTA cmd
+  inbmd -->> dispQ : OTA cmd
+  dispQ -->> dispQ : parse OTA cmd <br/> updater-type
+  dispQ --> ota : OTA cmd
+  ota ->> isv : update_tool_cmd <args>
+  isv ->> ota : status <OK/ERROR>
+  ota --> inbmd: status <OK/ERROR, msg>
+  inbmd -->> inbc : status <OK/ERROR, msg>
+  inbc ->> sudo : status <OK/ERROR, msg>
+
+```
+
+## Extensibility and Integration
+
+Extensibility in INBM's context can be defined by providing hooks in place to extend support:
+
+- connecting to a new device management server (dms), e.g. Amazon or Googles device management solutions
+  - this would involve adding new adapter in `cloud-client` which adhears to the protocol supported by the dms.
+
+- executing new OTA cmd type, to enable a customer's specific usecase for e.g. install drivers or run specific applications
+  - adding a new OTA cmd typically will involve adding new handlers in:
+    - `cloud-client` - additional handler for the new cmd
+    - `inbm-daemon` - additional logic to spawn a new type of ota thread
+    - `new-thread` - buisness logic executing the new cmd and reporting result
+
+- sending additional telemetry from device, e.g. GPU utilization
+  - add data collection routin in `telemerty-reporter`
+  - add `key:value` pairs for the new telemetry data getting collected
+  - possible update in `cloud-client` to send this data to `dms`
+
```

Comment:
What about our platform extendibility?  Where we have the configuration file to easily add the configuration properties to support FOTA on different platforms without a code change.

Reply:
Agree. Will add.